src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -48,4 +96,79 @@ class ObservationStore(
 
     return row.id!!
   }
+
+  fun updateObservationState(observationId: ObservationId, newState: ObservationState) {
+    requirePermissions {
+      if (newState == ObservationState.Completed) {
```

Comment:
Wondering if this should this somehow be `newState == ObservationState.Completed && oldState == ObservationState.InProgress`? If I'm understanding correctly that a user needs fewer permissions to complete an observation than to otherwise manage it.

Reply:
`oldState == ObservationState.Overdue` is also valid. The idea here was that a regular user can complete an observation (the permission check) from any state where transitioning to "Completed" is a valid transition (which is checked separately, since it also applies to super-admins).

Reply:
Gotcha, thanks